### DISCUSSION TITLE
Step-3 and step-4 (NBRA workflow with cp2k-xTB)

### DISCUSSION THREAD

**Discussion by Anjali-6-Ganai** [Quantum-Dynamics-Hub/libra-code#173] [Q&A] [answered]
I'm confused why is init_state=10 taken in https://github.com/compchem-cybertraining/Tutorials_Libra/blob/master/6_dynamics/2_nbra_workflows/9_step4_cp2k/tutorial.ipynb . What is init_state 10? To what excitation does init_state 10 belong? 

 In https://github.com/compchem-cybertraining/Tutorials_Libra/blob/master/6_dynamics/2_nbra_workflows/8_step3_cp2k/2_xTB/tutorials.ipynb , we had conisered 100 excitations . In section 3.2 , i feel that [10, 11], 'alp'] is the homo to lumo transition. If this is so, then init_state (in step 4) should be 91 as it is the 91th element in the list of 100 exciatations considered. Am I thinking the right way?

Another possibility I have in my mind, is [[[1, 11], 'alp'], [[1, 12], 'alp'], [[1, 13], 'alp'], [[1, 14], 'alp'], [[1, 15], 'alp'], [[1, 16], 'alp'], [[1, 17], 'alp'], [[1, 18], 'alp'], [[1, 19], 'alp'], [[1, 20], 'alp'] corresponds to the 1st excited state. [[2, 11], 'alp'], [[2, 12], 'alp'], [[2, 13], 'alp'], [[2, 14], 'alp'], [[2, 15], 'alp'], [[2, 16], 'alp'], [[2, 17], 'alp'], [[2, 18], 'alp'], [[2, 19], 'alp'], [[2, 20], 'alp'] corresponds to the 2nd excited state and so [[10, 11], 'alp'], [[10, 12], 'alp'], [[10, 13], 'alp'], [[10, 14], 'alp'], [[10, 15], 'alp'], [[10, 16], 'alp'], [[10, 17], 'alp'], [[10, 18], 'alp'], [[10, 19], 'alp'], [[10, 20], 'alp']] corresponds to the 10th excited state and it is this 10th excited state that we are studying in step 4 by specifying init_state=10. Is this the way?

Actually, I'm confused about the state index and which state we are considering the decay dynamics. It would be helpful if you please clarify this aspect.

**Comment by MohammadShakiba** [comment, marked as answer]
Hi Anjali, 

The init_state 10 in the dynamics means the dynamics starts from the 10th state in the excited state basis you built in step3. For example, in step3 if you have set the number of occupied orbitals to 5 and number of unoccupied orbitals to 4, then Libra will make a excited state basis set including all the excitation from the occupied orbitals to unoccupied orbitals which counts to 21 (including the ground state). In step3 input file, based on the sorting type you have defined, for example 'energy'-ordered, Libra will sort these excitation based on their energies and compute the NACs between them. 

For your 2nd question, you show the sd_unique_basis, which are the unique SDs that were built and they are not sorted. In fact, the HOMO to LUMO transition has index 1 (first excited state) in the Hvib matrices (note that the ground state index is 0). The reason is that when sorting by energy, this is the lowest excitation energy possible. For the next one (second excited state), it is either HOMO-1 to LUMO or HOMO to LUMO+1 depending on the energies of the orbitals and so on for other excited states. So, the 91st excited state is the 91st element of the energy-sorted SDs which can be any of the above excitation (we do not know with the above information what exactly it is but we know it is one of them).

**Comment by Anjali-6-Ganai** [comment]
Thank you very much for the clarification.
